lib/FrontendTool/FrontendTool.cpp

```diff
@@ -324,7 +325,7 @@ static void debugFailWithCrash() {
 
 /// Performs the compile requested by the user.
 /// \returns true on error
-static bool performCompile(CompilerInstance &Instance,
+static bool performCompile(std::unique_ptr<CompilerInstance> &Instance,
```

Comment:
Please update the doc comment about which conditions `Instance` will be reset under. (Honestly, maybe we should just pass a boolean to handle the diagnostics verification case. Checking that way down in the function is pretty subtle.)